compiler/resolution/visibleFunctions.cpp

```diff
@@ -479,297 +479,238 @@ void getVisibleFunctions(const char*      name,
-          // If we're not already in a use chain, by definition we can see
-          // private import.  If we're in a use chain, assume that private
-          // imports are not available to us
-          if (!inUseChain || !import->isPrivate) {
-            // Not all import statements define symbols for unqualified access,
-            // traverse into those that do when the name we're seeking is
-            // specified
-            if (import->skipSymbolSearch(name) == false) {
-              SymExpr* se = toSymExpr(import->src);
-
-              INT_ASSERT(se);
-              ModuleSymbol* mod = toModuleSymbol(se->symbol());
-              INT_ASSERT(mod);
-              if (mod->isVisible(call) == true) {
-                if (import->isARenamedSym(name) == true) {
-                  getVisibleFunctions(import->getRenamedSym(name), call,
-                                      mod->block, visited, visibleFns, true);
-                } else {
-                  getVisibleFunctions(name, call, mod->block, visited,
-                                      visibleFns, true);
-                }
-              }
+        }
+      }
+    } else if (ImportStmt* import = toImportStmt(expr)) {
+      if (needToTraverseUse(firstVisit, inUseChain, import->isPrivate)) {
+        // Not all import statements define symbols for unqualified access,
+        // traverse into those that do when the name we're seeking is
+        // specified
+        if (import->skipSymbolSearch(name) == false) {
+          SymExpr* se = toSymExpr(import->src);
+
+          INT_ASSERT(se);
+          ModuleSymbol* mod = toModuleSymbol(se->symbol());
+          INT_ASSERT(mod);
+          if (mod->isVisible(call)) {
+            if (import->isARenamedSym(name)) {
+              getVisibleFunctions(import->getRenamedSym(name), call,
+                                  mod->block, visited, visibleFns, true);
+            } else {
+              getVisibleFunctions(name, call, mod->block, visited,
+                                  visibleFns, true);
             }
           }
-
-        } else {
-          INT_FATAL("Expected ImportStmt or UseStmt");
         }
       }
+    } else {
+      INT_FATAL("Expected ImportStmt or UseStmt");
     }
+  }
+}
 
-    if (block != rootModule->block) {
-      BlockStmt* next  = getVisibilityScopeNoParentModule(block);
-
-      // Recurse in the enclosing block
-      getVisibleFunctions(name, call, next, visited, visibleFns, inUseChain);
+static void getVisibleFunctions(const char*           name,
+                                CallExpr*             call,
+                                BlockStmt*            block,
+                                std::set<BlockStmt*>& visited,
+                                Vec<FnSymbol*>&       visibleFns,
+                                bool                  inUseChain)
+{
+  const bool firstVisit = (visited.find(block) == visited.end());
 
-      if (instantiationPt != NULL) {
```

Comment:
The scope of this removed `if` is different than the scope of the same check that was removed on line 767.  I think that means that if the block being visited for the first time is the rootModule's block, we previously didn't check the instantiation point but now will do so.  I don't know what the implications of that are (and maybe the new behavior isn't a bad thing), but I suspect it was an edge case we didn't have a test for so would like to make sure you're aware of it since the PR message claims the business logic is unchanged.